src/PVRIptvData.cpp

```diff
@@ -1482,3 +1487,28 @@ int PVRIptvData::GetChannelId(const char * strChannelName, const char * strStrea
 
   return abs(iId);
 }
+
+std::string& PVRIptvData::AddHeaderToStreamUrl(std::string& strLine, const std::string& header)
```

Comment:
Why not pass the strLine as a const url and just capture the separator char before the return?

So with a signature of:
`std::string PVRIptvData::AddHeaderToStreamUrl(const std::string& url, const std::string& header)`

Reply:
And then a `char separator = '|';` which get's used in the return.

Reply:
Yes, can be done. Now the stream it's not corrected, so not edited: it's the right one.
For the signature i would keep strLine to be more clear, even if not strictly necessary (opinion).
Sadly the char separator depends about there are or not other headers in the stream line.

Reply:
Ok, but then why return anything at all? Just modify strLine OR return a value. Don’t do both.

Also your function is called add header to stream url but it’s argument is called strLine. Variable names should be relevant to the local scope, not the parent scope.

Reply:
Yeah, just saw it.